plugins/woocommerce/includes/wc-page-functions.php

```diff
@@ -168,7 +168,7 @@ function wc_nav_menu_item_classes( $menu_items ) {
 			$menu_id = (int) $menu_item->object_id;
 
 			// Unset active class for blog page.
-			if ( $page_for_posts === $menu_id ) {
+			if ( $page_for_posts === $menu_id && 'page' === $menu_item->object ) {
```

Comment:
Let's add a defensive check here before we access the object (I don't see it being accessed anywhere else, so we don't know if it will necessarily be set)
```suggestion
			if ( $page_for_posts === $menu_id && isset( $menu_item->object ) && 'page' === $menu_item->object ) {
```